### PR TITLE
[#16094] Messaging - In Chat - Implement Options Menu for 1-1 and com…

### LINKED ISSUE #16094: Messaging - In Chat -  Implement Options Menu for 1-1 and communties
Current version of the In Chat Messaging does not include any functionality for the options button.

Implement this as these options are available on other screens.

see red square box in image:

This should be implemented for the different variations, e,g 1-1, Group and Communities.
The individual actions can be implemented in follow up if this is too large or consider breaking this story into 3 separate tasks for each variation

Designs are here:

1-1: https://www.figma.com/file/7KIYbhoqNGAIFonE0w9TDz/Messages-for-Mobile?type=design&node-id=7365-512935&t=a5CFBvO6ImJoqxLC-4

Communities: https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=6301-422452&t=sHzSqc7SUJQNVVgm-4 

For communities a lot of these options already exist in: 
https://github.com/status-im/status-mobile/blob/develop/src/status_im2/contexts/communities/actions/community_options/view.cljs

A similar approach could be taken or this functionality could be reused/extended

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #16094

### CODE REVIEW

File: src/status_im2/contexts/chat/messages/view.cljs [hunk 1]
```diff
@@ -27,17 +28,18 @@
...
-                              (rf/sub [:contacts/contact-by-address chat-id]))
-        photo-path          (rf/sub [:chats/photo-path chat-id])
-        avatar-image-key    (if (seq (:images contact))
-                              :profile-picture
-                              :ring-background)]
+                chat-type]
+         :as   chat}     (rf/sub [:chats/current-chat])
+        display-name     (if (= chat-type constants/one-to-one-chat-type)
+                           (first (rf/sub [:contacts/contact-two-names-by-identity chat-id]))
+                           (str emoji " " chat-name))
```
Comment (J-Son89) [inline src/status_im2/contexts/chat/messages/view.cljs]: nit - should this be an i18n label? 🤔

**Review by J-Son89 — APPROVED** [review]
nice one @flexsurfer ! 🏄

File: src/status_im2/contexts/chat/messages/view.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im2/contexts/chat/messages/view.cljs]: its not my code, im not sure what's that, but doesn't look like label

**Comment by pavloburykh** [comment]
@flexsurfer thanx for the PR. Please take a look at the following issues

### ISSUE 1 "No matching clause" exception on opening options menu in community channel

**Steps:**

1. Open community channel
2. Tap on Options button
3. Observe the result

**Actual result:** app crashes with "No matching clause" exception. UI freezes, app needs to be restarted. 


https://github.com/status-im/status-mobile/assets/97245802/2cad8da2-cbda-418d-b5a8-2b8f2986d955

[Status-debug-logs (32).zip](https://github.com/status-im/status-mobile/files/11626812/Status-debug-logs.32.zip)

**Comment by pavloburykh** [comment]
### ISSUE 2 Keyboard is not closed when Options bottom sheet is opened in chat

**Steps:** 

1. Open chat
2. Make input field active to open keyboard
3. Tap options button
4. See if keyborad is getting closed when options bottom sheet appears 

**Actual result:** keyboard is not closed. On Android bottom sheet appears above the keyboard, on IOS behind it. 

**Android:** 

https://github.com/status-im/status-mobile/assets/97245802/511d4971-7bcc-4b9f-a9de-3dc6e442768a

**IOS:**

https://github.com/status-im/status-mobile/assets/97245802/1f97f429-ab75-4897-8c88-7bdae7a61598

**Comment by pavloburykh** [comment]
### ISSUE 3 Deleting chat via Option menu does not result in closing currently opend chat

**Steps:** 

1. Open 1-1 chat
2. Open Options menu
3. Select Delete chat option
4. Observe the result after deletion

**Expected result:** user is redirected to Messages screen. Chat is deleted. 

**Actual result:** user remains on broken chat screen with "Add unknown to send message" info message. 


https://github.com/status-im/status-mobile/assets/97245802/7e2ee173-b7ac-4221-8842-2336e0bc8ae2

**Comment by flexsurfer** [comment]
thank @pavloburykh fixed, for community chat i disabled options, should be implemented separately, we need to file separate issue with detailed description

**Review by siddarthkay — APPROVED** [review]

**Comment by pavloburykh** [comment]
@flexsurfer thank you. PR is ready to be merged. Failed e2e are not PR related.

**Pull request merged**